### PR TITLE
[codemod] markDynamoStrictTest batch 18

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117609
* #117606
* #117605
* __->__ #117604

[codemod] markDynamoStrictTest functorch/test_vmap
[codemod] markDynamoStrictTest profiler/test_profiler_tree
[codemod] markDynamoStrictTest profiler/test_profiler
[codemod] markDynamoStrictTest profiler/test_memory_profiler
[codemod] markDynamoStrictTest functorch/test_ops
[codemod] markDynamoStrictTest functorch/test_aotdispatch

### CODE REVIEW

File: torch/testing/_internal/dynamo_test_failures.py [hunk 1]
```diff
@@ -3178,6 +3172,123 @@
...
+    "TestTorchTidyProfiler.test_tensor_lists",  # profiler/test_profiler
+    "TestTorchTidyProfiler.test_pointers_and_ids",  # profiler/test_profiler
+    "TestTorchTidyProfiler.test_nnmodule_params",  # profiler/test_profiler
+    "TestTorchTidyProfiler.test_tensorimpl_invalidation_set",  # profiler/test_profiler
+    "TestProfiler.test_source_multithreaded_basic_work_in_main_thread_False",  # profiler/test_profiler
+    "TestTorchTidyProfiler.test_scalar_ins",  # profiler/test_profiler
+    "TestProfiler.test_profiler_op_event_args",  # profiler/test_profiler
+    "TestProfiler.test_source_multithreaded_complex_work_in_main_thread_True",  # profiler/test_profiler
+    "TestProfiler.test_source_multithreaded_multiple_preexisting_work_in_main_thread_True",  # profiler/test_profiler
+    "TestAOTAutograd.test_input_mutation_aliases_and_output_alias",  # functorch/test_aotdispatch
```
Comment (bdhirsh) [inline torch/testing/_internal/dynamo_test_failures.py:3235]: Hmm does this mean that dynamo is running directly on the `test_aotdispatch.py` tests? Do we want that? (having dynamo run on the AOTAutograd code seems.. fragile)

**Review by bdhirsh — APPROVED** [review]
left a q but stamping anyway

File: torch/testing/_internal/dynamo_test_failures.py (hunk 1 above)
Comment (zou3519) [reply torch/testing/_internal/dynamo_test_failures.py:3235]: Answered offline, but Dynamo should graph break correctly (or handle them) when it sees things like make_fx, FunctionalTensor, FakeTensor, etc.

**Review by malfet — APPROVED** [review]

**Comment by zou3519** [comment]
re-landed in another PR

**Pull request closed without merging**